python/lsst/sims/catUtils/utils/alertGenerator.py

```diff
@@ -0,0 +1,1475 @@
+        work.
+        """
+        return len(self._htmid_dict[htmid])
+
+    def obs_from_htmid(self, htmid):
+        """
+        Return a numpy array containing all of the ObservationMetaData
+        that interasect the trixel specified by htmid.
+
+        Must run subdivide_obs in order for this method to
+        work.
+        """
+        return self._obs_list[self._htmid_dict[htmid]]
+
+    def _output_alert_data(self, conn, data_cache):
+        """
+        Write a cache of alert data to the sqlite file currently open.
+
+        Parameters
+        ----------
+        conn is the connection to the sqlite file (already open)
+
+        data_cache is a dict containing all of the data to be written.
+        It will keyed on a string like 'i_j' where i is the obshistID
+        of an OpSim pointing and j is an arbitrary integer.  That key
+        will lead to another dict keyed on the columns being output to
+        the sqlite file.  The values of this second layer of dict are
+        numpy arrays.
+
+        Returns
+        -------
+        The number of rows written to the sqlite file
+        """
+
+        cursor = conn.cursor()
+        n_rows_0 = cursor.execute('SELECT COUNT(uniqueId) FROM alert_data').fetchall()
+
+        chunk_lengths = np.zeros(len(data_cache))
+
+        for i_cache_tag, cache_tag in enumerate(data_cache):
+            obsHistID = int(cache_tag.split('_')[0])
+            n_obj = len(data_cache[cache_tag]['uniqueId'])
+            chunk_lengths[i_cache_tag] = n_obj
+
+            values = ((int(data_cache[cache_tag]['uniqueId'][i_obj]),
+                      obsHistID,
+                      data_cache[cache_tag]['xPix'][i_obj],
+                      data_cache[cache_tag]['yPix'][i_obj],
+                      int(data_cache[cache_tag]['chipNum'][i_obj]),
+                      data_cache[cache_tag]['dflux'][i_obj],
+                      data_cache[cache_tag]['SNR'][i_obj],
+                      np.degrees(data_cache[cache_tag]['raICRS'][i_obj]),
+                      np.degrees(data_cache[cache_tag]['decICRS'][i_obj]))
+                      for i_obj in range(n_obj))
+            cursor.executemany('INSERT INTO alert_data VALUES (?,?,?,?,?,?,?,?,?)', values)
+        conn.commit()
+
+        n_rows_1 = cursor.execute('SELECT COUNT(uniqueId) FROM alert_data').fetchall()
+        conn.commit()
+        n_written = (n_rows_1[0][0]-n_rows_0[0][0])
+
+        return n_written
+
+    def _filter_on_chip_name_then_photometry(self, chunk, column_query,
+                                             obs_valid_dex, expmjd_list,
+                                             photometry_catalog):
+        """
+        Determine which simulated observations are actually worth storing
+        by first figuring out which land on an LSST detector and then
+        figuring out which satisfy the photometric thresholds for
+        observability and alert-worthiness.
```

Comment:
I did write them both because I was uncertain which would be faster.  I kept this one because I wasn't sure that I wouldn't find a database table that was more rapidly simulated this way than the other.  That has not proven to be true.  I will remove it in one commit. That should be easy enough to revert if I discover that I need this method, after all.